shared/rakefiles/xk.rake

```diff
@@ -0,0 +1,131 @@
+task :configure_serviceaccount => [@serviceaccount_key_file]
+rule @serviceaccount_key_file => [@gcp_creds_file] do
+  # TODO: This command is duplicated from exekube's gcp-project-init (and
+  # hardcodes 'projectowner' instead of $SA_NAME which is only defined in
+  # gcp-project-init). If gcp-project-init becomes idempotent (GPII-2989,
+  # https://github.com/exekube/exekube/issues/92), or if this 'keys create'
+  # step moves somewhere else in exekube, call this command from that place
+  # instead.
+  sh "gcloud iam service-accounts keys create $TF_VAR_serviceaccount_key \
+        --iam-account projectowner@$TF_VAR_project_id.iam.gserviceaccount.com"
+end
+
+@kubectl_creds_file = "/root/.kube/config"
+task :configure_kubectl => [@kubectl_creds_file]
+rule @kubectl_creds_file => [@gcp_creds_file] do
+  # This duplicates information in terraform code, 'k8s-cluster'
+  cluster_name = 'k8s-cluster'
+  # This duplicates information in terraform code, 'zone'. Could be a variable with some plumbing.
+  zone = 'us-central1-a'
+  sh "
+    existing_cluster=$(gcloud container clusters list --filter #{cluster_name} --zone #{zone} --project #{ENV["TF_VAR_project_id"]})
+    if [ $? == 0 ] && [ \"$existing_cluster\" != \"\" ]; then \
+      gcloud container clusters get-credentials #{cluster_name} --zone #{zone} --project #{ENV["TF_VAR_project_id"]}
+    fi"
+end
+
+# This task is being called from entrypoint.rake and runs inside exekube container.
+# It applies infra, secret-mgmt, sets secrets, and then executes arbitrary command from args[:cmd].
+# You should not invoke this task directly!
+task :xk, [:cmd, :skip_infra, :skip_secret_mgmt] => [@serviceaccount_key_file, @kubectl_creds_file] do |taskname, args|
+  @secrets = Secrets.collect_secrets()
+
+  sh "#{@exekube_cmd} up live/#{@env}/infra" unless args[:skip_infra]
+  sh "#{@exekube_cmd} up live/#{@env}/secret-mgmt" unless args[:skip_secret_mgmt]
+
+  Secrets.set_secrets(@secrets)
+
+  sh_filter "#{@exekube_cmd} #{args[:cmd]}" if args[:cmd]
+end
+
+task :infra_init => [@gcp_creds_file] do
+  # Steps to initialize GCP with a minimum set of resources to allow Terraform
+  # create the rest of the infrastructure.
+  # These steps are the same found in this tutorial:
+  # https://cloud.google.com/community/tutorials/managing-gcp-projects-with-terraform
+  #
+  # Due to the needed permissions for creating the following resources, only an
+  # administrator of the organization can run this task.
+
+  output = `#{@exekube_cmd} gcloud projects list --format='json' --filter='name:#{ENV["TF_VAR_project_id"]}'`
+  hash = JSON.parse(output)
+  if hash.empty?
+    puts "#{ENV["TF_VAR_project_id"]} not found, I'll create a new one"
+    sh "
+      #{@exekube_cmd} gcloud projects create #{ENV["TF_VAR_project_id"]} \
+      --organization #{ENV["ORGANIZATION_ID"]} \
+      --set-as-default"
+  else
+    sh "#{@exekube_cmd} gcloud config set project #{ENV["TF_VAR_project_id"]}"
```

Comment:
Can you use `:configure_current_project` for this?

Reply:
the file `xk.rake` is the only rake file used inside the Docker container, the task `:configure_current_project` is defined in the `entrypoint.rake`. Do you think that is worth to merge these two files?. If so, perhaps it could be addressed in another Jira task.